kyuubi-hive-beeline/src/main/java/org/apache/hive/beeline/KyuubiCommands.java

```diff
@@ -82,9 +82,8 @@ private boolean sourceFileInternal(File sourceFile) throws IOException {
     BufferedReader reader = null;
     try {
       reader = new BufferedReader(new FileReader(sourceFile));
-      String extra = reader.readLine();
-      String lines = null;
-      while (extra != null) {
+      String lines = null, extra;
+      while ((extra = reader.readLine()) != null) {
         if (beeLine.isComment(extra)) {
```

Comment:
comment line will cause infinite loop.

Reply:
cc @turboFei

Reply:
[HIVE-16693](https://issues.apache.org/jira/browse/HIVE-16693): beeline "source" command freezes if you have a comment in it

Fix Version: 3.0.0